src/hivpy/sexual_behaviour.py

```diff
@@ -364,18 +366,21 @@ def update_long_term_partners(self, population):
 
         (self.ltp_balance_factor[SexType.Male],
          self.ltp_balance_factor[SexType.Female]) = self.balance_ltp_factor(population)
+
         # new relationships
+        population.data[col.LTP_AGE_GROUP] = np.digitize(
+            population.data[col.AGE], self.new_ltp_age_bins)
         new_relationships = population.transform_group(
-            [col.SEX, col.AGE],  self.start_new_ltp, sub_pop=partnerless_idx)
+            [col.SEX, col.LTP_AGE_GROUP],  self.start_new_ltp, sub_pop=partnerless_idx)
         population.data.loc[partnerless_idx, col.LONG_TERM_PARTNER] = new_relationships
-        # not sure about indexing on this next line
-        # We only want to update LTP_LONGEVITY where LONG_TERM_PARTNER is true within
-        # the sub-population defined by partnerless_idx
+
+        # calculate longevity for new relationships this time step
+        # (but not for existing relationships)
         new_ltp_subpop = population.data.index[start_partnerless &
                                                population.data[col.LONG_TERM_PARTNER]]
         longevity = population.transform_group(
-            [col.AGE, col.SEX], self.new_ltp_longevity, sub_pop=new_ltp_subpop)
-        population.data.loc[new_ltp_subpop, col.LTP_LONGEVITY] = longevity
+            [col.LTP_AGE_GROUP], self.new_ltp_longevity, sub_pop=new_ltp_subpop)
+        population.data.loc[new_ltp_subpop, col.LTP_LONGEVITY] = longevity.astype(int)
         # ending relationships
         population.data.loc[partnered_idx, col.LONG_TERM_PARTNER] = population.transform_group(
-            [col.AGE, col.LTP_LONGEVITY], self.continue_ltp, sub_pop=partnered_idx)
+            [col.LTP_LONGEVITY], self.continue_ltp, sub_pop=partnered_idx)
```

Comment:
Taking this as an opportunity to highlight that this may be worth coding as a higher-level method on the `Population`: it appears twice in this method alone, and I think it would be more readable. Something like
```python
population.update_column_by_groups(
    column_name=col.LONG_TERM_PARTNER,
    group_by=[col.LTP_LONGEVITY],
    apply=self.continue_ltp,
    sub_pop=partnered_idx
)
```
that calls `transform_group` and assigns the result to the specified column.